src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -227,8 +233,21 @@ export class Kernel implements IKernel {
         }
         traceInfo(`Restart requested ${this.notebookDocument.uri}`);
         this.startCancellation.cancel();
-        await this.kernelExecution.restart(this._notebookPromise);
-        traceInfoIfCI(`Restarted ${this.notebookDocument.uri}`);
+        try {
+            await this.kernelExecution.restart(this._notebookPromise?.then((item) => item.session));
+            traceInfoIfCI(`Restarted ${this.notebookDocument.uri}`);
+        } catch (ex) {
```

Comment:
lol, found a bug, if the kernel is dead, we don't really handle it well.
Now, i'm logging the errors and killing the underlying notebook/jupytersession

Reply:
Came across this while looking at the weird kernel issues we've been having & the kernel failures Jim has reported as well (basically he couldn't start a kernel)